lib/internal/policy/manifest.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+const {
+  ERR_MANIFEST_ASSERT_INTEGRITY,
+  ERR_MANIFEST_INTEGRITY_MISMATCH,
+  ERR_MANIFEST_UNKNOWN_ONERROR,
+} = require('internal/errors').codes;
+const SRI = require('internal/policy/sri');
+const { SafeWeakMap } = require('internal/safe_globals');
+const crypto = require('crypto');
+const { Buffer } = require('buffer');
+const { URL } = require('url');
+const { createHash, timingSafeEqual } = crypto;
+const Hash$update = Function.call.bind(crypto.Hash.prototype.update);
+const Hash$digest = Function.call.bind(crypto.Hash.prototype.digest);
+const Buffer$equals = Function.call.bind(Buffer.prototype.equals);
+const Buffer$toString = Function.call.bind(Buffer.prototype.toString);
+const RegExp$test = Function.call.bind(RegExp.prototype.test);
+const entries = Object.entries;
+const INTEGRITIES = new SafeWeakMap();
+const REACTIONS = new SafeWeakMap();
+const RELATIVE_URL_STRING_PATTERN = /^\.{0,2}\//;
+const { shouldAbortOnUncaughtException } = process.binding('config');
+const { abort, exit, _rawDebug } = process;
+function REACTION_THROW(error) {
+  throw error;
+}
+function REACTION_EXIT(error) {
+  REACTION_LOG(error);
+  if (shouldAbortOnUncaughtException) {
+    abort();
+  }
+  exit(1);
```

Comment:
From what I understand this could be triggered in the middle of some user code execution (from any `require` call) where some async hooks or domain handlers, etc may have been setup, or some Immediate has been scheduled in the C++ layer. Does this take any of that into account during the shutdown? Do the users have a chance to clean up any resources during this shutdown process? If not this should be pointed out in the docs.

Reply:
I would not expect this to allow user code to run, we can call it out in the docs. If there is a desire to run user code I'd be curious on the reason to allow it and why to not use `throw` instead.

Reply:
added to doc